src/config_store.c

```diff
@@ -92,9 +108,56 @@ void ConfigStore_Init(ConfigStore *p)
     p->_fd = -1;
 }
 
+/*To delete the leftover tmp files on the device startup*/
+void DeleteFileHelper(struct dirent *file, char *filePath)
+{
+    char *ptr, *fileName;
+    fileName = file->d_name;
+
+    //rindex() is a string handling function that returns a pointer to the last occurrence
+    //of character c in string s, or a NULL pointer if c does not occur in the string.
+    ptr = rindex(fileName, '.');
+
+    //Check for filename extensions
+    if ((ptr != NULL) && (strncmp(ptr, ".tmp", 4) == 0))
```

Comment:
>4 [](start = 47, length = 1)

This needs to be `5` to account for the null-terminator and do a strict check for `.tmp`, otherwise a file name ending in `.tmprtr` or similar will still pass this test.

Reply:
Changed to 5.